okhttp/src/jvmMain/kotlin/okhttp3/internal/platform/Jdk9Platform.kt

```diff
@@ -37,6 +38,10 @@ open class Jdk9Platform : Platform() {
 
     sslParameters.applicationProtocols = names.toTypedArray()
 
+    if (hostname != null) {
+      sslParameters.serverNames = listOf(SNIHostName(hostname))
```

Comment:
My read is we haven’t done SNI correctly on non-Java platforms. . . . forever? Wow!